xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp

```diff
@@ -154,14 +154,12 @@ bool CDVDDemuxClient::ParsePacket(DemuxPacket* pkt)
     int len = stream->m_parser->parser->split(stream->m_context, pkt->pData, pkt->iSize);
     if (len > 0 && len < FF_MAX_EXTRADATA_SIZE)
     {
-      if (st->ExtraData)
-        delete[] st->ExtraData;
       st->changes++;
       st->disabled = false;
       st->ExtraSize = len;
-      st->ExtraData = new uint8_t[len+AV_INPUT_BUFFER_PADDING_SIZE];
-      memcpy(st->ExtraData, pkt->pData, len);
-      memset(st->ExtraData + len, 0 , AV_INPUT_BUFFER_PADDING_SIZE);
+      st->ExtraData = std::unique_ptr<uint8_t[]>(new uint8_t[len + AV_INPUT_BUFFER_PADDING_SIZE]);
```

Comment:
Can you use `std::make_unique<uint8[]>()` here and below? It removes the exposed pointer from using `new`, and because it performs the allocation it can make sure everything is allocated together.

Reply:
@garbear, you are right.

Reply:
The commit is updated.